state-transition/core/core_test.go

```diff
@@ -90,29 +90,31 @@ func buildNextBlock(
 	root := beaconState.HashTreeRoot()
 	parentBlkHeader.SetStateRoot(root)
 
+	// build the block
+	blk, err := types.NewBeaconBlockWithVersion(
+		parentBlkHeader.GetSlot()+1,
+		parentBlkHeader.GetProposerIndex(),
+		parentBlkHeader.HashTreeRoot(),
+		version.Deneb1(),
+	)
+
 	// build the payload
 	payload := &types.ExecutionPayload{
+		Versionable:   blk,
 		Timestamp:     timestamp,
 		ExtraData:     []byte("testing"),
 		Transactions:  [][]byte{},
 		Withdrawals:   withdrawals,
 		BaseFeePerGas: math.NewU256(0),
-		EpVersion:     version.Deneb1(),
 	}
 	parentBeaconBlockRoot := parentBlkHeader.HashTreeRoot()
 	ethBlk, _, err := types.MakeEthBlock(payload, &parentBeaconBlockRoot)
 	require.NoError(t, err)
 	payload.BlockHash = common.ExecutionHash(ethBlk.Hash())
 
-	// finally build the block
-	blk, err := types.NewBeaconBlockWithVersion(
-		parentBlkHeader.GetSlot()+1,
-		parentBlkHeader.GetProposerIndex(),
-		parentBlkHeader.HashTreeRoot(),
-		version.Deneb1(),
-	)
 	require.NoError(t, err)
 	blk.Body = &types.BeaconBlockBody{
+		Versionable:      blk,
```

Comment:
technically now, it is an anti-pattern to use the "parent" struct as the `Versionable` now. Instead each type should have its own "base" `versionable` struct. Like define it once and set it in the `ExecutionPayload` and `BeaconBlockBody`. 

Now normally  `NewBeaconBlockWithVersion` already sets the Versionable field correctly, but since in this test we are overwriting the `ExecutionPayload` and `BeaconBlockBody` structs entirely, it becomes necessary to set explicitly.